external/loaders/tests/test__batch.py

```diff
@@ -79,6 +82,25 @@ def test_batches_from_mapper(mapper):
             assert len(dim) == original_dim_lengths[dim]
 
 
+def test_load_batches_with_catalog_path(mapper, datadir):
```

Comment:
Thanks for updating this test as we discussed offline.  Could you also add a brief comment describing the intent of this test for any future readers of the code?  It may not be immediately obvious why the `KeyError` indicates the custom catalog was used (we mainly tested things this way out of convenience and the fact that if loading the batches succeeded it would be indistinguishable from using the default catalog).